homeassistant/components/withings/common.py

```diff
@@ -0,0 +1,308 @@
+            _LOGGER.info("Looks like the service is available again")
+            WithingsDataManager.service_available = True
+            return True
+
+    async def call(self, function, is_first_call=True, throttle_domain=None):
+        """Call an api method and handle the result."""
+        throttle_data = self.get_throttle_data(throttle_domain)
+
+        should_throttle = (
+            throttle_domain and throttle_data and not throttle_data.is_expired()
+        )
+
+        try:
+            if should_throttle:
+                _LOGGER.debug("Throttling call for domain: %s", throttle_domain)
+                result = throttle_data.data
+            else:
+                _LOGGER.debug("Running call.")
+                result = await self._hass.async_add_executor_job(function)
+
+                # Update throttle data.
+                self.set_throttle_data(
+                    throttle_domain, ThrottleData(self.get_throttle_interval(), result)
+                )
+
+            WithingsDataManager.print_service_available()
+            return result
+
+        except TokenUpdated:
+            WithingsDataManager.print_service_available()
+            if not is_first_call:
+                raise ServiceError(
+                    "Stuck in a token update loop. This should never happen"
+                )
+
+            _LOGGER.info("Token updated, re-running call.")
+            return await self.call(function, False, throttle_domain)
+
+        except MissingTokenError as ex:
+            raise NotAuthenticatedError(ex)
+
+        except Exception as ex:  # pylint: disable=broad-except
+            # Service error, probably not authenticated.
+            if NOT_AUTHENTICATED_ERROR.match(str(ex)):
+                raise NotAuthenticatedError(ex)
+
+            # Probably a network error.
+            WithingsDataManager.print_service_unavailable()
+            raise PlatformNotReady(ex)
+
+    async def check_authenticated(self):
+        """Check if the user is authenticated."""
+
+        def function():
+            return self._api.request("user", "getdevice", version="v2")
+
+        return await self.call(function)
+
+    async def update_measures(self):
+        """Update the measures data."""
+
+        def function():
+            return self._api.get_measures()
+
+        self._measures = await self.call(function, throttle_domain="update_measures")
+
+        return self._measures
+
+    async def update_sleep(self):
+        """Update the sleep data."""
+        end_date = int(time.time())
+        start_date = end_date - (6 * 60 * 60)
+
+        def function():
+            return self._api.get_sleep(startdate=start_date, enddate=end_date)
+
+        self._sleep = await self.call(function, throttle_domain="update_sleep")
+
+        return self._sleep
+
+    async def update_sleep_summary(self):
+        """Update the sleep summary data."""
+        now = datetime.datetime.utcnow()
```

Comment:
Please use our datetime utils as much as possible. There's `utcnow` in `util/dt.py`.